src/sdk/map_client.js

```diff
@@ -557,6 +563,52 @@ class MapClient {
             }));
     }
 
+    async move_blocks_to_storage_class() {
+        // Block movement is not done if move_to_tier was not requested or if insufficient
+        // previous tier information is provided.
+        if (!this.move_to_tier || !this.current_tiers || this.current_tiers.length !== this.chunks.length) return;
```

Comment:
I think the last check is more of an error
```suggestion
        if (!this.move_to_tier || !this.current_tiers) return;
        if (this.current_tiers.length !== this.chunks.length) throw new Error('...');
```